.github/workflows/scripts/ghstack-perm-check.py

```diff
@@ -152,19 +159,91 @@ def rebase(self):
                 f"User {actor} must be a maintainer {self.REPO} to rebase someone else's PR",
             )
 
+
+class GHStack(ChatOps):
+    def __init__(self, session, event):
+        super().__init__(session, event)
+        self.pr_numbers = self.ghstack_pr_numbers()
+
+    def get_pr_ref(self):
+        return self.PR["head"]["ref"].replace("/head", "/orig")
+
+    def ghstack_pr_numbers(self):
+        proc = subprocess.Popen(
+            "git log FETCH_HEAD...$(git merge-base FETCH_HEAD origin/main)",
+            stdout=subprocess.PIPE,
+            shell=True,
+        )
+        out, _ = proc.communicate()
+        self.must(proc.wait() == 0, "`git log` command failed!")
+
+        pr_numbers = re.findall(
+            r"Pull Request resolved: https://github.com/.*?/pull/([0-9]+)",
+            out.decode("utf-8"),
+        )
+        pr_numbers = list(map(int, pr_numbers))
+        self.must(
+            pr_numbers and pr_numbers[0] == self.NUMBER,
+            "Extracted PR numbers not seems right!",
+        )
+        return pr_numbers
+
+    def land(self):
+        # Enforce requirements for every PR in the stack
+        for n in self.pr_numbers:
+            # Checking Approvals
+            self.land_prerequisites(n)
+
+        print(":: All PRs are ready to be landed!")
+
+    def rebase(self):
+        self.rebase_prerequisites()
         print(":: All PRs are ready to be rebased!")
 
 
+class GitHub(ChatOps):
+    def get_pr_ref(self):
+        return f"pull/{self.NUMBER}/head"
+
+    def must(self, cond, msg):
+        must(self.gh, self.REPO, self.NUMBER, cond, msg)
+
+    def land(self):
+        # Enforce requirements for the PR
+        self.land_prerequisites()
+        print(":: The PR is ready to be landed!")
+
+    def rebase(self):
+        self.rebase_prerequisites()
+        print(":: The PR is ready to be rebased!")
+
+
 def main(check):
     # Setup the wrapper class
     EV = json.loads(sys.stdin.read())
-    ghstack_checks = GHStackChecks(os.environ["GITHUB_TOKEN"], EV)
+    session = get_session(os.environ["GITHUB_TOKEN"])
+    chatops = None
+    isgh = is_ghstack(session, EV)
+
+    if isgh:
+        chatops = GHStack(session, EV)
+    else:
+        chatops = GitHub(session, EV)
+
+    # Set a number of useful environment variables
+    env_file = os.getenv("GITHUB_ENV")
+    with open(env_file, "a") as envfile:
+        envfile.write(f"IS_GHSTACK={isgh}\n")
+        envfile.write(f"REPO={chatops.REPO}\n")
+        envfile.write(f"PR_NUMBER={chatops.NUMBER}\n")
+        envfile.write(f"PR_REF={chatops.pr_ref}\n")
+        envfile.write(f"PR_URL={chatops.PR_URL}\n")
```

Comment:
I don't understand what this is supposed to be doing, so tried locally, and it's not any clearer. How does this work?

```
In [4]: import os

In [5]: env_file = os.getenv("BAR")

In [6]: x=True

In [7]: with open(env_file, "a") as envfile:
   ...:     envfile.write(f"FOO={x}\n")
   ...:

In [8]: env_file
Out[8]: 'x'

In [9]: os.getenv("BAR")
Out[9]: 'x'

In [10]: os.getenv("FOO")
```

Even if that did work in python, I believe env variables are only set for the current process, and it seems like you're using these values outside the python process. did you test this behavior?